docs/proposals/unified-backend-architecture/tech-spec.md

```diff
@@ -0,0 +1,239 @@
+   internet connection.
+
+These problems make impossible to install such a heavyweight client on a mobile device. The purpose of Unified
+Backend Architecture is to solve these problems.
+
+We propose to create a lightweight Daedalus client. And it is possible to unify the different use cases under a
+single architecture, starting from the crucial observation that both interfacing with an hardware wallet or
+provide a mobile client can be seen as a specialisation of the general case of having a "keyless" wallet backend
+operating with the private keys and cryptography "hosted" on the client side. It will allow to keep blockchain
+on a backend only.
+
+## Prerequisites
+
+The reader should be familiar with following concepts:
+
+1. Daedalus frontend,
+2. wallet backend web API,
+3. work with wallets,
+4. transaction processing.
+
+Also it is assumed that the reader knows basic terminology from [Cardano Docs](https://cardanodocs.com/),
+for example, UTXO.
+
+## Assumptions
+
+N/A, see **Prerequisites**.
+
+## Requirements
+
+Unification of the wallet backend architecture should allow to:
+
+1. work with different hardware wallets in the same manner,
+2. create lightweight Daedalus mobile clients (iOS/Android),
+3. create lightweight Daedalus desktop clients (Windows/macOS/Linux).
+
+The main benefit of such a lightweight client is independence from the growing blockchain: the client does not
+store blockchain locally and does not wait for a "Syncing blocks".
+
+## Key Idea
+
+As mentioned before, the key idea is to unify the different use cases under a single architecture, we mean
+hardware wallets, mobile clients and lightweight desktop clients.
+
+In the case of [Ledger](https://www.ledgerwallet.com/) (or any other hardware wallet) the privte keys and the
+cryptography are hosted on the device, whereas in a mobile client they are held on a smartphone. And this schema
+fits for "featherweight Daedalus desktops" that do not want to host the growing blockchain on the computer's
+storage.
+
+More specifically, a Daedalus mobile client will consist of an iOS or Android application which will communicate
+with a cluster of nodes that will keep the global UTXO state for the blockchain and that will be capable
+to reconstruct each mobile wallet state without hosting any user-specific data.
+
+## Multi-Tenant Architecture
+
+Mobile clients will communicate with a clusters of nodes, so the basic requirement is an ability of one node to work
+with particular number of clients simultaneously. For example, if 1 cluster has to serve 10000 mobile clients and
+this cluster consists of 100 nodes, each node has to be able to work with 100 clients simultaneously and reliably.
+Thus, the longest procedure is a new payment (creating new transaction), and even if all these 100 clients will
+sign their transactions and send it for publishing at the same time, the node must be able to handle these
+transactions reliably and as fast as possible.
```

Comment:
Yes, though actually it's the medium term average tx rate here that's important, short term fluctuations can be handled by queues/buffers. So there's the sustained tx submission rate and the queue size (to handle activity bursts).